bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/handler/ChromecastHandler.java

```diff
@@ -88,26 +88,31 @@ public void initialize() {
         ChromecastConfig config = getConfigAs(ChromecastConfig.class);
 
         final String ipAddress = config.ipAddress;
-        if (ipAddress == null || ipAddress.isEmpty()) {
+        if (ipAddress == null || ipAddress.isBlank()) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
                     "Cannot connect to Chromecast. IP address is not valid or missing.");
             return;
         }
 
-        Coordinator localCoordinator = coordinator;
-        if (localCoordinator != null && (!localCoordinator.chromeCast.getAddress().equals(ipAddress)
-                || (localCoordinator.chromeCast.getPort() != config.port))) {
-            localCoordinator.destroy();
-            localCoordinator = coordinator = null;
-        }
+        updateStatus(ThingStatus.UNKNOWN);
 
-        if (localCoordinator == null) {
-            ChromeCast chromecast = new ChromeCast(ipAddress, config.port);
-            localCoordinator = new Coordinator(this, thing, chromecast, config.refreshRate, audioHTTPServer,
-                    callbackUrl);
-            localCoordinator.initialize();
-            coordinator = localCoordinator;
-        }
+        scheduler.submit(() -> {
+            // initialize connection to devices asynchronously
+            Coordinator localCoordinator = coordinator;
+            if (localCoordinator != null && (!localCoordinator.chromeCast.getAddress().equals(ipAddress)
+                    || localCoordinator.chromeCast.getPort() != config.port)) {
+                localCoordinator.destroy();
+                localCoordinator = coordinator = null;
+            }
+
+            if (localCoordinator == null) {
+                ChromeCast chromecast = new ChromeCast(ipAddress, config.port);
+                localCoordinator = new Coordinator(this, thing, chromecast, config.refreshRate, audioHTTPServer,
+                        callbackUrl);
+                localCoordinator.initialize();
+                coordinator = localCoordinator;
+            }
+        });
```

Comment:
I think quickly cycling the handler (initialize->dispose->initialize) would cause multiple Coordinator instances to be created simultaneously and also lose track of one of them.

Reply:
Okay. Calling the `coordinator.initialize()` method asynchronously and keep the instance creating like before should work either.

Reply:
Only if `coordinator.initialize()` is synchronized along with `coordinator.destroy()` as well as adding an early exit in `coordinator.initialize()` if `coordinator.destroy()` was called first.

I think that would make it threadproof.

Reply:
@cweitkamp Do you have this PR still on your radar?

Reply:
@kaikreuzer Thanks for the reminder. I updated the PR with the suggested solution above.